OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -190,6 +193,10 @@ protected void ScheduleDelayedAction(int t, Action a)
 		// The world coordinate model uses Actor.Orientation
 		public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 		{
+			attack = attackTraits.FirstOrDefault(Exts.IsTraitEnabled);
+			if (attack == null)
+				return null;
```

Comment:
only small thought- this is probably correct, though might have some (good) impact in case of no enabled Attack trait if that is normal.